src/core/parsers/base.py

```diff
@@ -14,40 +14,65 @@ class BaseParser(object):
     Base parser. Can parse basic data ids for wikidata
     """
 
+    def _is_valid_numeric_id_range(self, value):
+        for num in [int(num) for num in filter(None, re.split("[QMPLFS\-]", value))]:
```

Comment:
Your function is checking if "any" number is within the range. Don't you mean to ensure that all entity numbers need to pass?

In case it is: here is the one-liner that replaces your loop:

```
return all([ 0 < int(num) < 2147483648 for num in re.findall(r"[QMPLFS](\d+)", value)])
```